dbms/src/Flash/Pipeline/Schedule/TaskQueues/FIFOQueryIdCache.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <common/types.h>
+
+#include <mutex>
+#include <unordered_set>
+
+namespace DB
+{
+class FIFOQueryIdCache
```

Comment:
I'm a little confused that is the `id` in this cache the `MPPTaskId` or `MPPGatherId` or `MPPQueryId`?

Reply:
I think for the Pipeline Model, a PipelineExecutor is a "query",
Just like the difference between tidb query and mpp query.
So the id here is the id of the PipelineExecutor.

Reply:
For mpptask, the PipelineExecutor's id is mpptaskid.
But currently there is no id of cop/batchCop, so a null value is given